### PR TITLE
part of parallel c++

### PR DESCRIPTION
**Pull request by alan-stokes**
the current impression is that this improves the performance of data out. so pr-ed so that the rest of data out don't lose it. 

depends upon this being merged first due to being built upon it.
- [ ] spinnaker tools: https://github.com/SpiNNakerManchester/spinnaker_tools/pull/71

### CODE REVIEW

**Comment by lplana** [comment]
I'm afraid I don't think that this should be merged as is.

1. Why bring in the 'mon_delegate' branch changes here? There is a separate pull request for that branch and this will just confuse issues.
2. I don't understand the changes to the 'setup' file. There is no indication as to why several lines were commented out.
3. I understand that the intention of the proposed modification is to leave the decision to throw away a mailbox message in the hands of the application code instead of the monitor. This relies on careful application coding, which is usually dangerous. The application can deadlock if it doesn't timeout or can run out of mailbox buffers if not paying attention.

**Comment by alan-stokes** [comment]
hi luis. 

why 1. id guess that due to spinnman being tied to it, then new changes for the tools can be based off it, this doesn't mean this has to be merged before mon deligate has already been done. ill add the dependency for clarity. (plus i didn't know where it sat, just that it was needed for the speed up).

2 & 3 i guess @lucaperes could explain, as i'm not taking ownership of this PR, just putting it into the space of the Data out project so that its changes are not lost, if indeed they are needed at all is a separate discussion that im not putting a hand in.

**Comment by rowleya** [comment]
I think these changes were experimental - more work should be done to avoid the issues raised.

File: scamp/scamp-isr.c [hunk 1]
```diff
@@ -264,18 +264,18 @@ INT_HANDLER ap_int()
...
 
 	sdp_msg_t *msg = sark_msg_get();
 
 	if (msg != NULL) {
+		vcpu->mbox_mp_cmd = SHM_IDLE;
 	    sark_msg_cpy(msg, shm_msg);
 	    msg_queue_insert(msg, 0);
+	    sark_shmsg_free(shm_msg);
 	} else {
 	    sw_error(SW_OPT);
```
Comment (rowleya) [inline scamp/scamp-isr.c:276]: A possible simple option is to add a vcpu->mbox_mp_cmd = SHM_ERR here, which will then tell the sender that something has changed at least.

**Comment by lucaperes** [comment]
> I think these changes were experimental


I confirm that

File: setup [hunk 2]
```diff
@@ -10,9 +10,9 @@ set -a
 #-------------------------------------------------------------------------------
 
 # This file should be sourced, not executed!
-if [ "$0" == "${BASH_SOURCE[0]}" ]; then
-  echo "WARNING: $0 should be sourced, not executed. Try: source $0"
-fi
+#if [ "$0" == "${BASH_SOURCE[0]}" ]; then
+#  echo "WARNING: $0 should be sourced, not executed. Try: source $0"
+#fi
```
Comment (dkfellows) [inline setup:15]: Why are these three lines commented out?

Comment (dkfellows) [reply setup:15]: The protection against being run as a stand-alone script (which won't work; it's a script whose purpose is to set environment variables for other scripts) is highly valuable. I see no good reason to remove it.

**Review by dkfellows — CHANGES_REQUESTED** [review]
Just one comment here; it looks like something done to a bit of (non-obvious) setup code was actually wrong.

I'd be totally for someone providing better options for the setup code too in the non-GCC case.

**Comment by alan-stokes** [comment]
killing as the correct thing is here https://github.com/SpiNNakerManchester/spinnaker_tools/pull/73

**Pull request closed without merging**